test/gtest/lstm_extra.cpp

```diff
@@ -0,0 +1,133 @@
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+auto GetTestCases(std::string precision)
+{
+    std::string flags       = "test_lstm --verbose " + precision;
+    std::string commonFlags = " --batch-size 32 --seq-len 3 --batch-seq 32 32 32 --vector-len 128 "
+                              "--hidden-size 128 --num-layers 1 --in-mode 0 --bias-mode 0";
+
+    // clang-format off
+    return std::vector<std::string>{
+        {flags + commonFlags + " -dir-mode 0 --no-hx"},
+        {flags + commonFlags + " -dir-mode 0 --no-dhy"},
+        {flags + commonFlags + " -dir-mode 0 --no-hx --no-dhy"},
+        {flags + commonFlags + " -dir-mode 0 --no-cx"},
+        {flags + commonFlags + " -dir-mode 0 --no-hx --no-cx"},
+        {flags + commonFlags + " -dir-mode 0 --no-dcy"},
+        {flags + commonFlags + " -dir-mode 0 --no-cx --no-dcy"},
+        {flags + commonFlags + " -dir-mode 1 --no-hx"},
+        {flags + commonFlags + " -dir-mode 1 --no-dhy"},
+        {flags + commonFlags + " -dir-mode 1 --no-hx --no-dhy"},
+        {flags + commonFlags + " -dir-mode 1 --no-cx"},
+        {flags + commonFlags + " -dir-mode 1 --no-hx --no-cx"},
+        {flags + commonFlags + " -dir-mode 1 --no-dcy"},
+        {flags + commonFlags + " -dir-mode 1 --no-cx --no-dcy"},
+        {flags + commonFlags + " -dir-mode 0 --no-hy"},
+        {flags + commonFlags + " -dir-mode 0 --no-dhx"},
+        {flags + commonFlags + " -dir-mode 0 --no-hy --no-dhx"},
+        {flags + commonFlags + " -dir-mode 0 --no-cy"},
+        {flags + commonFlags + " -dir-mode 0 --no-hy --no-cy"},
+        {flags + commonFlags + " -dir-mode 0 --no-dcx"},
+        {flags + commonFlags + " -dir-mode 0 --no-cy --no-dcx"},
+        {flags + commonFlags + " -dir-mode 1 --no-hy"},
+        {flags + commonFlags + " -dir-mode 1 --no-dhx"},
+        {flags + commonFlags + " -dir-mode 1 --no-hy --no-dhx"},
+        {flags + commonFlags + " -dir-mode 1 --no-cy"},
+        {flags + commonFlags + " -dir-mode 1 --no-hy --no-cy"},
+        {flags + commonFlags + " -dir-mode 1 --no-dcx"},
+        {flags + commonFlags + " -dir-mode 1 --no-cy --no-dcx"},
+	{flags + commonFlags + " -dir-mode 0 --no-hx --no-dhy --no-cx --no-dcy --no-hy --no-dhx --no-cy --no-dcx"},
+	{flags + commonFlags + " -dir-mode 1 --no-hx --no-dhy --no-cx --no-dcy --no-hy --no-dhx --no-cy --no-dcx"}
+    };
+    // clang-format on
+}
+
+using TestCase = decltype(GetTestCases({}))::value_type;
+
+class ConfigWithFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+class ConfigWithHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+bool IsTestSupportedForDevice()
+{
+    using namespace miopen::debug;
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::Default>;
+    return ::IsTestSupportedForDevMask<d_mask, e_mask>();
+}
+
+void Run2dDriver(miopenDataType_t prec)
+{
+    if(!IsTestSupportedForDevice())
```

Comment:
```suggestion
    if(!(IsTestSupportedForDevice() //
        && (miopen::IsUnset(ENV(MIOPEN_TEST_ALL)) // standalone run
            || miopen::IsEnabled(ENV(MIOPEN_TEST_ALL))))) // or full tests enabled
```
the test should be run if supported by device AND (standalone run OR full tests enabled).